core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -938,7 +938,7 @@ class KafkaApis(val requestChannel: RequestChannel,
 
       var unconvertedFetchResponse: FetchResponse = null
```

Comment:
This is the root cause of the error-prone :(

Reply:
I understand, mutable variables are the root of many issues. We should keep variables as immutable as possible to avoid unexpected errors.